Sources/WalletConnectSign/Sign/SignClient.swift

```diff
@@ -157,16 +162,16 @@ public final class SignClient {
         namespaces: [String: SessionNamespace]
     ) throws {
         //TODO - accounts should be validated for matching namespaces BEFORE responding proposal
-        guard let (sessionTopic, proposal) = pairingEngine.approveProposal(proposerPubKey: proposalId, validating: namespaces) else {return}
+        let (sessionTopic, proposal) = try approveEngine.approveProposal(proposerPubKey: proposalId, validating: namespaces)
         try sessionEngine.settle(topic: sessionTopic, proposal: proposal, namespaces: namespaces)
```

Comment:
One of the motivations to have an approveEngine for me was to not pass `(sessionTopic, proposal)` between engines in the client.
a think the `sessionEngine.settle` method should also go to approveEngine
to me `proposal response` and `settle request` are like two pieces of an approval so make sense to perform them in one object.